app/graphql/foreman_graphql_schema.rb

```diff
@@ -20,14 +20,22 @@ def self.object_from_id(id, query_ctx)
     return unless id.present?
 
     _, type_name, item_id = Foreman::GlobalId.decode(id)
-    model_class = type_name.safe_constantize
+    type_class = "::Types::#{type_name}".safe_constantize
+
+    model_class = type_class&.model_class
 
     return unless model_class
 
     RecordLoader.for(model_class).load(item_id.to_i)
   end
 
-  def self.resolve_type(_, obj, _)
-    types[obj.class.name]
+  def self.resolve_type(_, object, _)
+    klass = object.class
+    if klass.respond_to?(:graphql_type) && klass.graphql_type
```

Comment:
```suggestion
    if klass.try(:graphql_type)
```
Although if you add it to ApplicationRecord, would there be a case where it doesn't respond to `graphql_type`?

Reply:
Yes, the object could be something different than a descendant of `ApplicationRecord `. E.g. Networks from a Compute Resource or something like this.